src/Elliptic/Systems/Poisson/Equations.cpp

```diff
@@ -23,59 +23,31 @@
-void ComputeFirstOrderOperatorAction<Dim>::apply(
-    const gsl::not_null<Scalar<DataVector>*> operator_for_field_source,
+void euclidean_fluxes(
     const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
-        operator_for_auxiliary_field_source,
-    const tnsr::i<DataVector, Dim, Frame::Inertial>& grad_field,
-    const tnsr::I<DataVector, Dim, Frame::Inertial>& auxiliary_field,
-    const Mesh<Dim>& mesh,
-    const InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
-        inverse_jacobian) noexcept {
-  auto div_vars =
-      make_with_value<Variables<tmpl::list<Tags::AuxiliaryField<Dim>>>>(
-          auxiliary_field, 0.);
-  get<Tags::AuxiliaryField<Dim>>(div_vars) = auxiliary_field;
-  // Tensors don't support math operations yet, so we have to `get` the
-  // DataVector for the sign flip
-  get(*operator_for_field_source) =
-      -1. * get(get<::Tags::div<Tags::AuxiliaryField<Dim>>>(
-                divergence(div_vars, mesh, inverse_jacobian)));
+        flux_for_field,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& field_gradient) noexcept {
   for (size_t d = 0; d < Dim; d++) {
-    operator_for_auxiliary_field_source->get(d) =
-        grad_field.get(d) - auxiliary_field.get(d);
+    flux_for_field->get(d) = field_gradient.get(d);
   }
 }
 
 template <size_t Dim>
-void ComputeFirstOrderNormalDotFluxes<Dim>::apply(
-    const gsl::not_null<Scalar<DataVector>*> normal_dot_flux_for_field,
-    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
-        normal_dot_flux_for_auxiliary_field,
-    const Scalar<DataVector>& field,
-    const tnsr::I<DataVector, Dim, Frame::Inertial>& auxiliary_field,
-    const tnsr::i<DataVector, Dim, Frame::Inertial>&
-        interface_unit_normal) noexcept {
-  // The minus sign in the equation is canceled by the one in `lift_flux`
-  get(*normal_dot_flux_for_field) =
-      get<0>(interface_unit_normal) * get<0>(auxiliary_field);
-  for (size_t d = 1; d < Dim; d++) {
-    get(*normal_dot_flux_for_field) +=
-        interface_unit_normal.get(d) * auxiliary_field.get(d);
-  }
-
-  // The minus sign is to cancel the one in `lift_flux`
+void auxiliary_fluxes(gsl::not_null<tnsr::IJ<DataVector, Dim, Frame::Inertial>*>
+                          flux_for_gradient,
+                      const Scalar<DataVector>& field) noexcept {
+  *flux_for_gradient =
+      make_with_value<tnsr::Ij<DataVector, Dim, Frame::Inertial>>(field, 0.);
   for (size_t d = 0; d < Dim; d++) {
-    normal_dot_flux_for_auxiliary_field->get(d) =
-        -interface_unit_normal.get(d) * get(field);
+    flux_for_gradient->get(d, d) = get(field);
   }
 }
 
 template <size_t Dim>
 void FirstOrderInternalPenaltyFlux<Dim>::package_data(
     const gsl::not_null<Variables<package_tags>*> packaged_data,
     const Scalar<DataVector>& field,
-    const tnsr::i<DataVector, Dim, Frame::Inertial>& grad_field,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& grad_field,
```

Comment:
Can you please explain why this change? I thought in a previous commit in this PR you changed the gradient to a lower index. In terms of math, I would have thought the gradient of a field would be a convector

Reply:
This quantity is now the principal symbol applied to the gradient, i.e. for the Euclidean Poisson system it's just the gradient with its index raised with the flat metric. I didn't change the variable name because for the Poisson system it's still the same values and I want to keep changes to the flux to a minimum. That's because I'll delete this Poisson-specific implementation of the flux in #1671 and replace it with a generic internal penalty flux for elliptic systems.